src/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -429,6 +429,80 @@ public static bool TrySkipBlock(BinaryReader reader)
 
             return true;
         }
+
+        public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
+        {
+            const int OffsetToFilenameLength = 22; // from the point after the signature
+
+            if (reader.ReadUInt32() != SignatureConstant)
+                return false;
+
+            if (reader.BaseStream.Length < reader.BaseStream.Position + OffsetToFilenameLength)
```

Comment:
You may want to think a bit more about what checks you do for the size.  TrySkipBlock did this check because it was going to seek past it.  You aren't doing an explicit seek of `OffsetToFilenameLength` but instead you are doing fixed-sized reads and then seeks based on the data you've read. I'd also expect comments that indicate what information you are seeking past/seeking-to.

You're missing the check on fileNameLength / extra-FieldLength.

As a whole, try to think about garunteeing you have enough data available to proceed and when you read an offset that isn't fixed size make sure you check up front that you can seek to that offset.

Reply:
Sure , will review all again and add needed check